### ISSUE TITLE
Powers that are fractions break on negative numbers

### ISSUE THREAD

**Issue by OAGr** [getguesstimate/guesstimate-app#978]
`` = (-3)^1.2`` and similar break.

**Comment by philastrophist** [comment]
Could this ever be fixed without using rejection sampling?

**Comment by OAGr** [comment]
Nothing comes to mind. Hypothetically we could do something fancy like MCMC, but this would be quite tricky.

**Comment by philastrophist** [comment]
Using something like https://chi-feng.github.io/mcmc-demo/app.html?algorithm=HamiltonianMC&target=banana I assume? Rejection sampling will be really inefficient sometimes. 
The main problem with global mcmc is that if you specify that a = -1 to 2 and then do b = sqrt(a), the distribution of a no longer matches the input. 

I think, to maintain consistency, we need to be able to propagate invalid values along the chain rather than throwing a sampling error. 
Then the summary graphs will ignore them. The input specification will then match the output. 

 Some cases will simply mark all values as invalid, so the user should be able to specify the number of samples upfront (but maybe a different error type is required if all the recurved  samples are invalid - “too much information lost”?). 

I guess this is just rejection sampling under a different name but at least it won’t crash anything without notice. It’s probably the easiest solution to implement as well. 

TLDR:
1. User adjustable nsamples
2. Propagate invalid samples (summaries should ignore them in calculation but show how many there were)
3. New error type for the case where all(samples == invalid) in order to be transparent to the user

What do you think?

**Comment by OAGr** [comment]
That sounds quite reasonable to me, thanks for the advice here. Happy to have this be the broad plan in this area.

That said, note that Guesstimate isn't having active development now. I'm planning to work to rev this up in the future, but it could be some time.

**Comment by philastrophist** [comment]
Fair enough, I don’t have any experience with js so I can’t help right now either

**Comment by OAGr** [comment]
Thanks for letting me know. I think the intersection of js experience + probability experience is pretty small :)

**Comment by philastrophist** [comment]
I’m a fan of this project though, because scripting up mcmc for simple questions is overkill and visualisation is great!
